app/code/core/Mage/Catalog/Model/Layer.php

```diff
@@ -115,10 +115,7 @@ public function prepareProductCollection($collection)
             ->addAttributeToSelect(Mage::getSingleton('catalog/config')->getProductAttributes())
             ->addPriceData()
             ->addTaxPercents()
-            ->addUrlRewrite($this->getCurrentCategory()->getId())
-            ->addAttributeToFilter('status', [
-                'in' => Mage::getSingleton('catalog/product_status')->getVisibleStatusIds()
-            ]);
```

Comment:
I don't think this code should be removed, the previous version of this file had

`Mage::getSingleton('catalog/product_status')->addVisibleFilterToCollection($collection);`

which was filtering the status (although it wasn't working)

Reply:
What is the benefit of your change?

There was no status filter before. Why to add this filter, that is removed later?

Reply:
maybe I'm missing something

Reply:
`addVisibleFilterToCollection()` did nothing (it just returns `$this`) ... so there was no filter on `status`. It was never used, and marked as deprecated from beginning ... maybe for some reason? Maybe b/c it was implemented in another way?

Your change to `Catalog/Model/Layer.php` added a filter that gets removed here:

```
 $excludedWherePart = Mage_Catalog_Model_Resource_Product_Collection::MAIN_TABLE_ALIAS . '.status'; 
 foreach ($wherePart as $key => $wherePartItem) { 
     if (strpos($wherePartItem, $excludedWherePart) !== false) { 
         $wherePart[$key] = new Zend_Db_Expr('1=1'); 
         continue; 
     } 
     $wherePart[$key] = $this->_replaceTableAlias($wherePartItem); 
 } 
```

For enabled flat product tables, status filter is set in `Mage_Catalog_Model_Resource_Product_Collection::_initSelect()` and thats gets replaced by `1=1` as first part of the WHERE condition. Nothing wrong so far. `1=1` is no ugly hack, its a comon pattern for dynamically created queriers, b/c you can add `AND/OR [condition]` w/o to care if it is first part in WHERE.

Also `setPart()` works. There is nothing lost. Before #2603 we had these part `"(e.status = 1)"`. After your change we have 

```
 0 = "(e.status = 1)"
 1 = "AND (e.status IN(1))" # this comes from your change
```

The "where type" `AND` is there (nothing that can get lost), but both are replaced with `1=1`, leading to wrong query.

I agree ... this is wrong (or at least not planned and covered to have `status` multiple times).

Again ...every  `e.status` gets removed intentionally, so it makes no sense to (re-)add it with `addAttributeToFilter('status', [ ....`